cups/file.c

```diff
@@ -335,6 +336,131 @@ _cupsFileCheckFilter(
 #endif /* !_WIN32 */
 
 
+/*
+ * 'cupsFileGetConfAndComments()' - Get line and comments from a configuration file.
+ */
+
+char *					/* O  - Line read or @code NULL@ on end of file or error */
+_cupsFileGetConfAndComments(cups_file_t *fp,	/* I  - CUPS file */
+                char        *buf,	/* O  - String buffer */
+		size_t      buflen,	/* I  - Size of string buffer */
+                char        **value,	/* O  - Pointer to value */
+		int         *linenum)	/* IO - Current line number */
+{
+  char	*ptr;				/* Pointer into line */
+
+
+ /*
+  * Range check input...
+  */
+
+  DEBUG_printf(("2cupsFileGetConfAndComments(fp=%p, buf=%p, buflen=" CUPS_LLFMT
+                ", value=%p, linenum=%p)", (void *)fp, (void *)buf, CUPS_LLCAST buflen, (void *)value, (void *)linenum));
+
+  if (!fp || (fp->mode != 'r' && fp->mode != 's') ||
+      !buf || buflen < 2 || !value)
+  {
+    if (value)
+      *value = NULL;
+
+    return (NULL);
+  }
+
+ /*
+  * Read the next line...
+  */
+
+  *value = NULL;
+
+  while (cupsFileGets(fp, buf, buflen))
+  {
+    (*linenum) ++;
+
+    /*
+     * Strip the comment from inline comment...
+     */
+     if ((ptr = strchr(buf, '#')) != NULL)
+     {
+       int index = (int) (ptr - buf);
+       for(int i=index-1; i>=0; i--)
+         if (!_cups_isspace(buf[i]))
+         {
+           buf[index] = '\0';
+           break;
+         }       
+     }
+
+     /*
+      * Strip leading whitespace...
+      */
+
+      for (ptr = buf; _cups_isspace(*ptr); ptr ++);
+
+      if (ptr > buf)
+        _cups_strcpy(buf, ptr);
+
+      /*
+       * See if there is anything left...
+       */
+
+       if (buf[0] != '#')
```

Comment:
We have to check whether there is anything in the buffer, and if so, check for # - this big if-else block is unnecessary and add unnecessary indentation - the good practice is to not nest the code too much, the code should be as straight as possible.

```
if (buf[0])
{
  // return comment if any
  if (buf[0] == '#')
    return ptr;

  /*
   * Otherwise grab any value and return...
   */
....
```